public/pages/DetectorResults/containers/AnomalyResults.tsx

```diff
@@ -80,21 +83,28 @@ export function AnomalyResults(props: AnomalyResultsProps) {
   const core = React.useContext(CoreServicesContext) as CoreStart;
   const dispatch = useDispatch();
   const detectorId = props.detectorId;
-  const detector = useSelector(
+  const detector = useSelector(    
     (state: AppState) => state.ad.detectors[detectorId]
   );
+  const dataSourceEnabled = getDataSourcePlugin().dataSourceEnabled;
+  const location = useLocation();
+  const dataSourceId = new URLSearchParams(location.search).get(DATA_SOURCE_ID) || '';
```

Comment:
thought about this idea, having a fn to parse url query params like what we do currently on List page. Thought it's fine because right now it's just one line of getting the dataSourceId. But I can make this a utility fn :)

Reply:
Great, can you do this? It will just make everything more maintainable and flexible when we have all of these common processes for handling MDS data. It also ensures consistency that datasourceId is getting initialized to the same value across all pages.

Reply:
can I address this in the upcoming pr? Already made the change, but it's in a different branch

Reply:
sure